genson/src/test/java/com/owlike/genson/ext/jackson/JacksonAnnotationsSupportTest.java

```diff
@@ -0,0 +1,448 @@
+                "  \"n\":\"Bilbo\"\n" +
+                "}";
+        assertEquals(new JsonPropertyOnGetterSetterCustomName("Bilbo"), genson.deserialize(input, JsonPropertyOnGetterSetterCustomName.class));
+    }
+
+    @Test
+    public void testJsonCreatorDeserialization() {
+        final String input = "{\n" +
+                "  \"@class\":\"com.owlike.genson.ext.jackson.JacksonAnnotationsSupportTest$JsonCreator\",\n" +
+                "  \"name\":\"Bilbo\"\n" +
+                "}";
+        assertEquals(new JsonCreator("Bilbo"), genson.deserialize(input, JsonCreator.class));
+    }
+
+    @Test
+    public void testJsonCreatorFactoryMethodDeserialization() {
+        final String input = "{\n" +
+                "  \"@class\":\"com.owlike.genson.ext.jackson.JacksonAnnotationsSupportTest$JsonCreatorFactoryMethod\",\n" +
+                "  \"name\":\"Bilbo\"\n" +
+                "}";
+        assertEquals(JsonCreatorFactoryMethod.newInstance("Bilbo"), genson.deserialize(input, JsonCreatorFactoryMethod.class));
+    }
+
+    // ---- Test Types ------------------------------------------------------
+
+    public static class JsonPropertyOnField {
+
+        @JsonProperty
+        private String name;
+
+        @SuppressWarnings("unused")
+        public JsonPropertyOnField() {}
+        JsonPropertyOnField(final String name) {
+            this.name = name;
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) return true;
+            if (!(o instanceof JsonPropertyOnField)) return false;
+            final JsonPropertyOnField that = (JsonPropertyOnField) o;
+            return Objects.equals(name, that.name);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(name);
+        }
+    }
+
+    public static class JsonPropertyOnGetterSetter {
+
+        private String name;
+
+        @SuppressWarnings("unused")
+        public JsonPropertyOnGetterSetter() {}
+        JsonPropertyOnGetterSetter(final String name) {
+            this.name = name;
+        }
+
+        @JsonProperty
+        public String getName() {
+            return name;
+        }
+
+        @JsonProperty
+        public void setName(final String name) {
+            this.name = name;
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) return true;
+            if (!(o instanceof JsonPropertyOnGetterSetter)) return false;
+            final JsonPropertyOnGetterSetter that = (JsonPropertyOnGetterSetter) o;
+            return Objects.equals(name, that.name);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(name);
+        }
+    }
+
+    public static class JsonPropertyOnFieldIgnored {
+
+        @JsonProperty
+        @JsonIgnore
+        private String name;
```

Comment:
Should this even be allowed? Doesn't make much sense to have both `@JsonProperty` and `@JsonIgnore`, imo...